openapi.yml

```diff
@@ -0,0 +1,475 @@
+      parameters:
+        - name: executable
+          in: path
+          description: Name of the executable
+          schema: {$ref: '#/components/schemas/OcrdExecutable'}
+          required: true
+        - name: job-id
+          in: path
+          description: ID of the ProcessorJob
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return Log
+          content:
+            'text/plain': {}
+        '404': {content: {}, description: 'ProcessorJob not found'}
+    post:
+      tags: ['processing']
+      operationId: logProcessor
+      parameters:
+        - name: executable
+          in: path
+          description: LogEntry to log
+          schema: {$ref: '#/components/schemas/LogEntry'}
+          required: true
+        - name: job-id
+          in: path
+          description: ID of the ProcessorJob
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return Log
+          content:
+            'text/plain': {}
+        '404': {content: {}, description: 'ProcessorJob not found'}
+
+  '/workflow':
+    post:
+      tags: ['workflow', 'discovery']
+      operationId: postWorkflow
+      requestBody:
+        description: 'Register a new workflow'
+        content:
+          'text/vnd.ocrd+sh': {}
+        required: true
+      responses:
+        '200':
+          description: Created a new OCR-D workflow
+          content: {application/json: {schema: {$ref: '#/components/schemas/Workflow'}}}
+
+  '/workflow/{workflow-id}':
+    put:
+      tags: ['workflow']
+      operationId: putWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      requestBody:
+        description: 'Replace existing or create new workflow'
+        content:
+          'text/vnd.ocrd+sh': {}
+        required: true
+      responses:
+        '200':
+          description: Created a new OCR-D workflow
+          content: {application/json: {schema: {$ref: '#/components/schemas/Workflow'}}}
+    get:
+      tags: ['workflow', 'discovery']
+      operationId: getWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return ProcessorJob
+          content:
+            application/json: {schema: {$ref: '#/components/schemas/Workflow'}}
+            application/vnd.ocrd+zip: {}
+        '404': {content: {}, description: 'Workflow not available'}
+    post:
+      tags: ['workflow']
+      operationId: runWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      responses:
```

Comment:
Currently we only have a `200` answer. What happens if the workflow fails at some point?

Reply:
`400` TODO

Reply:
In this case, the response would always be 201 unless the workflow is invalid detected by static analysis (e.g. missing required parameters etc.)

Reply:
b416666